### PR TITLE
Add presigned url compute methods to client

### PR DESCRIPTION
**Pull request by MrCreosote**
Also bump minio startup wait from 0.5 -> 1s, seeing rare failures

### CODE REVIEW

File: cdmtaskservice/s3/client.py [hunk 1]
```diff
@@ -204,3 +218,60 @@ async def head(client, buk=buk, key=key):  # bind the current value of the varia
...
+        raise ValueError(f"{name} is required")
+
+
+def _require_string(string, name):
+    if not string or not string.strip():
+        raise ValueError(f"{name} is required")
+    return string.strip()
+
+
+def _check_int(inte, name, minimum=1):
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply cdmtaskservice/s3/client.py:275]: I'm not sure what a reasonable minimum would be for expiration; it'd depend heavily on the use case. As such I just made the minimum the lowest number that makes any sense at all

File: cdmtaskservice/s3/client.py [hunk 2]
```diff
@@ -204,3 +218,60 @@ async def head(client, buk=buk, key=key):  # bind the current value of the varia
...
+    if inte < minimum:
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply cdmtaskservice/s3/client.py:276]: My philosophy on type checking on python is this:
* if it's user input, you need to type check
* If it's an internal method and has type hints, you assume the programmer using the method is abiding by the type hints and so you don't need to check.

Ideally we'd have mypy running to enforce point 2 but when I used it for the ID mapping service it was just too clunky and painful. I figure I'll try it again when you can just do `python --strict-types` or something.

That being said, I added type hints to the internal methods.

[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply cdmtaskservice/s3/client.py:276]: 👍

**Review by Tianhao-Gu — APPROVED** [review]
LGTM.

**Pull request merged**